bolparser.py

```diff
@@ -0,0 +1,118 @@
+__author__ = 'Amogh Jalihal'
+
+import json
+import re
+import sys
+
+class bolParser:
+    def __init__(self, bol):
+        self.bol = bol
+        self.Avartan = list()
+        self.Vibhaag = list()
+        self.Matra = list()
+        self.num_avartans = 0
+        self.taaldict = {}
+        
+        self.__cleanupBol()
+        self.__getTaalDict()
+        self.__getAvartan()
+        self.__getVibhaag()
+        self.__getMatra()
+        
+        self.num_avartans = len(self.Avartan)
+
+    def __cleanupBol(self):
+        # Replace multiple spaces in input with single space
+        self.bol = " ".join(self.bol.split())
+        # Remove padded | if any
+        self.bol = re.sub("[ ]*\|[ ]*","|",self.bol)
+
+    def __getTaalDict(self):
+        with open("data/taals.json",'r') as f:
+            self.taaldict = json.load(f)
+        
+    def __getAvartan(self):
+        self.Avartan = self.bol.split('||')[:-1]
+
+    def __getVibhaag(self):
+        self.Vibhaag = [av.split('|') for av in self.Avartan]
+
+    def __getMatra(self):
+        GroupedAkshars = [[v.split(' ') for v in V] for V in self.Vibhaag]
+        for av in GroupedAkshars:
```

Comment:
What's a, v and av in this case?

Reply:
v for vibhaag, av for avartan. Do you have a convention I can follow?